test/ranges-test.cc

```diff
@@ -742,3 +742,21 @@ TEST(ranges_test, movable_only_istream_iter_join) {
   EXPECT_EQ("1, 2, 3, 4, 5",
             fmt::format("{}", fmt::join(std::move(first), last, ", ")));
 }
+
+namespace {
+struct not_range {
+  void begin() const {}
+  void end() const {}
+};
+}  // namespace
+FMT_BEGIN_NAMESPACE
+template <> struct formatter<not_range> : formatter<string_view> {
+  auto format(const not_range&, format_context& ctx) const
+      -> format_context::iterator {
+    return formatter<string_view>::format("Dummy output", ctx);
+  }
+};
+FMT_END_NAMESPACE
+TEST(ranges_test, format_not_range_using_specialized_formatter) {
+  EXPECT_EQ("Dummy output", fmt::format("{}", not_range{}));
+}
```

Comment:
If you're satisfied with that, then I'll remove the test and this can just have the static assert then. AFAIK there isn't a gtest way to static assert so will just use a regular one

Reply:
You can keep the test and make a dynamic check (e.g. EXPECT_FALSE) instead of static_assert. Then at least some of the failures might be recoverable. But it's not important.

Reply:
Had pushed before your comment, apologies, but seems like failing as early as possible ie. compile time rather than runtime is fine anyway, so will leave it without the test if you're ok with that, can change it back if not